### PR TITLE
Airport - Week 5 Challenge

### PR DESCRIPTION
**Pull request by raphlevy**

### CODE REVIEW

File: spec/plane_spec.rb [hunk 1]
```diff
@@ -16,14 +16,47 @@
...
+  	expect(plane).to respond_to :can_take_off
+  end
+
+  it 'can take off' do
+  	plane = Plane.new "Paris"
+  	expect {plane.can_take_off}.to raise_error 'the plane cannot take off if it is not landed'
+  end
+
+  it 'is flying after take off' do
+  	plane = Plane.new "Paris"
```
Comment (raphlevy) [inline spec/plane_spec.rb]: this method is the same as 'is flying when created'. you maybe want to test that it is 'flying' after it landed and took off again.

File: lib/airport.rb [hunk 2]
```diff
@@ -1,2 +1,54 @@
...
+		@capacity = capacity.to_i
+	end
+
+	def airport 
+		@airport
+	end
+
+	def inst_take_off
+	end
+
```
Comment (danldb) [inline lib/airport.rb]: delete unused methods

File: lib/airport.rb [hunk 3]
```diff
@@ -1,2 +1,54 @@
+require_relative 'plane'
+
 class Airport
+
```
Comment (danldb) [inline lib/airport.rb]: Please fix your indentation to 2 spaces - this code is hard to read due to large indentation.

File: lib/airport.rb [hunk 4]
```diff
@@ -1,2 +1,54 @@
...
+		@airport.delete(plane)
+		plane.can_take_off
+		puts "Enter new destination:"
+		plane.destination = gets.chomp
+	end
+
+	def weather? 
+	arr = [:sunny, :stormy]
+	return arr[rand(2)] == :stormy	
+	end
```
Comment (danldb) [inline lib/airport.rb]: Only use a ? in a method name if you are returning a boolean.

File: lib/airport.rb [hunk 5]
```diff
@@ -1,2 +1,54 @@
...
+
+	def inst_take_off
+	end
+
+	def release_plane plane
+		#change weather to another word explaining the status
+		raise 'the weather does not allow to take off' if weather?
+		raise 'wrong location we cannot release this plane' if self.location != plane.destination
+		@airport.delete(plane)
```
Comment (danldb) [inline lib/airport.rb]: you should write an attr_reader for @airport - it is a code smell to see @ throughout your code.

File: lib/airport.rb [hunk 6]
```diff
@@ -1,2 +1,54 @@
...
+		raise 'wrong location we cannot accept this plane' if self.location != plane.destination
+		raise 'airport is full' if full?
+		plane.can_land
+		@airport.push plane
+	end
+
+	def full?
+		self.airport.size >= capacity
+	end
+
```
Comment (danldb) [inline lib/airport.rb]: no private methods - should all these methods be exposed?

File: lib/airport.rb [hunk 7]
```diff
@@ -1,2 +1,54 @@
...
+	def inst_landing 
+	end
+
+	def receive_plane plane
+		raise 'the weather does not allow to land' if weather?
+		raise 'wrong location we cannot accept this plane' if self.location != plane.destination
+		raise 'airport is full' if full?
+		plane.can_land
```
Comment (danldb) [inline lib/airport.rb]: tell don't ask - tell the plan it should land, and trust the plane to do it's job.

File: lib/plane.rb [hunk 8]
```diff
@@ -1,2 +1,28 @@
 class Plane
+
+	attr_accessor :destination
```
Comment (danldb) [inline lib/plane.rb]: why an attr_accesssor? Do you want another class to be able to change your plane's destination?

File: spec/airport_spec.rb [hunk 9]
```diff
@@ -13,21 +13,99 @@
...
+        #   plane = Plane.new "Paris" 
+        #   airport1 = Airport.new "Paris"
+        #   airport1.receive_plane plane
+        #   v = airport1.airport.length
+        #   expect(airport1.release_plane(plane).length).to eq (v - 1)
+        # end
+
+        #test plane taking off
+        #test get.chomp
+
```
Comment (danldb) [inline spec/airport_spec.rb]: always delete unused code

File: spec/plane_spec.rb [hunk 10]
```diff
@@ -16,14 +16,48 @@
...
+  end
+  
+
+  it 'function can_land exists' do
+   	plane = Plane.new "Paris"
+   	expect(plane).to respond_to :can_land
+	end
+
+	it 'can_land' do
+		plane = Plane.new "Paris"
```
Comment (danldb) [inline spec/plane_spec.rb]: use subject { Plane.new 'Paris' } instead of setting a local variable

File: spec/airport_spec.rb [hunk 11]
```diff
@@ -40,9 +118,21 @@
     # the plane can not land, and must not be in the airport
 
     context 'when weather conditions are stormy' do
-      xit 'does not allow a plane to take off'
+      it 'does not allow a plane to take off' do
+        plane = Plane.new "Paris"
```
Comment (danldb) [inline spec/airport_spec.rb]: use a double instead of a real plane

File: lib/airport.rb (hunk 4 above)
Comment (danldb) [inline lib/airport.rb]: indentation

File: lib/airport.rb [hunk 12]
```diff
@@ -1,2 +1,54 @@
...
+		#change weather to another word explaining the status
+		raise 'the weather does not allow to take off' if weather?
+		raise 'wrong location we cannot release this plane' if self.location != plane.destination
+		@airport.delete(plane)
+		plane.can_take_off
+		puts "Enter new destination:"
+		plane.destination = gets.chomp
+	end
+
+	def weather?
```
Comment (danldb) [inline lib/airport.rb]: extract this functionality into a module or class and inject that into the airport - that also will make it easier to stub.

**Comment by danldb** [comment]
:construction: 

I haven't given you a pass for this challenge - your classes are too tightly coupled, you need to pay attention to private vs public methods, and you have gone beyond the brief by adding destination.

**Pull request closed without merging**